### PR TITLE
modified atomic::ordering about batch_in_progress

### PR DESCRIPTION
**Pull request by wang384670111**
https://github.com/AleoHQ/snarkOS/blob/8cfa1657746bcdc372207f788f093b3060aac350/node/store/src/rocksdb/map.rs#L95
I think the use of ordering here is irregular. In the start_atomic function, batch_in_progress is not synchronized with other variables, so Relaxed can be used for batch_in_progress.

https://github.com/AleoHQ/snarkOS/blob/8cfa1657746bcdc372207f788f093b3060aac350/node/store/src/rocksdb/map.rs#L116
https://github.com/AleoHQ/snarkOS/blob/8cfa1657746bcdc372207f788f093b3060aac350/node/store/src/rocksdb/map.rs#L149
In the finish_atomic and abort_atomic functions, batch_in_progress is used to synchronize the atomic_batch variable. Therefore, the read operation of batch_in_progress needs to see the changes of the atomic_batch variable.  Using Ordering::SeqCst is unnecessary and may impact program performance, I think just Acquire/Release needs to be used here to ensure the correctness of the program.

### CODE REVIEW

**Comment by ljedrz** [comment]
Thanks! I think it makes sense and this "loosening" is something that has always been the plan; let me just go over it once more with a fresh set of eyes tomorrow :+1:.

**Comment by wang384670111** [comment]
> Thanks! I think it makes sense and this "loosening" is something that has always been the plan; let me just go over it once more with a fresh set of eyes tomorrow 👍.

Great! I also found some other improper usage of atomic types in snarkOS, and I will submit some PRs later.

**Comment by ljedrz** [comment]
Thanks for the further adjustments to atomic ordering!

I have a request: could you collect them all in a single pull request (it could be this one) and re-categorize them a little bit? The categories (which would be in their own commits) are:
- `batch_in_progress` (the same commit as in this PR)
- `shutdown` (collected from all the PRs that adjust it in different node types)
- `SEQUENTIAL_NODE_ID`
- `puzzle_instances`
- `block_generation_time`

In other words, in the end there would be a single pull request with 5 commits, each containing one of the listed categories, and outlining that the atomics that don't require synchronization are changed to `Relaxed`, and ones that do are changed to the `Acquire`-`Release` "sandwich".

**Comment by wang384670111** [comment]
> 

Sure! I will resubmit a new pull request that combines all the previous 5 pull requests uploaded.

**Comment by ljedrz** [comment]
Closing in favor of https://github.com/AleoHQ/snarkOS/pull/2317.
